### DISCUSSION TITLE
Add ability to register custom reporters

### DISCUSSION THREAD

**Discussion by llaville** [PHP-CS-Fixer/PHP-CS-Fixer#7995] [Feature ideas]
Hello,

Because I've worked on two reports (an improved version of standard `txt` format, and a new `sarif` format), I'm a bit sad to see that there is no ability yet to register it as it's case for custom fixers.

So today, I've worked on it, and I'm able to propose a solution (already available on my fork : see https://github.com/llaville/PHP-CS-Fixer/tree/features/add_custom_reporters)

This new feature is summarized like that :  https://github.com/llaville/PHP-CS-Fixer/blob/features/add_custom_reporters/doc/custom_reports.rst

E.g, may be invoke as standard command `vendor/bin/php-cs-fixer check src/SarifLog.php --format sarif`

With a `.php-cs-fixer.php` like this : 

```php
<?php

return (new PhpCsFixer\Config())
    ->registerCustomReporters([
        new \Bartlett\Sarif\Converter\PhpCsFixerConverter(),
    ])
    ->setFinder(
        PhpCsFixer\Finder::create()
            ->files()
            ->in(__DIR__ . '/src')
            ->name('SarifLog.php')
    );
```

WDYT about this new feature idea ?

**Comment by llaville** [comment]
Any answer will be greatly appreciated

**Reply by Wirone** [reply]
Any of public API modifications require @keradus, personally I am OK with adding this support.

**Reply by keradus** [reply]
i'm happy with idea.
didn't check implementatiaon
